python/social_bot/gazebo_agent.py

```diff
@@ -180,10 +195,37 @@ def get_full_states_observation(self, teacher):
         """
         task_specific_ob = teacher.get_task_specific_observation(self)
         agent_pose = np.array(self.get_pose()).flatten()
-        agent_vel = np.array(self.get_velocities()).flatten()
-        internal_states = self.get_internal_states()
-        obs = np.concatenate(
-            (task_specific_ob, agent_pose, agent_vel, internal_states), axis=0)
+        if self._use_simple_full_states:
+            # assumes GoalTask and that the first 3 dims of the
+            # task_specific_observation give the goal position.
+            yaw = agent_pose[5]
+            obs = []
+            while len(task_specific_ob) > 1:
+                x = task_specific_ob[0] - agent_pose[0]
+                y = task_specific_ob[1] - agent_pose[1]
+                # rotate -yaw
+                rotated_x = x * np.cos(-yaw) - y * np.sin(-yaw)
+                rotated_y = x * np.sin(-yaw) + y * np.cos(-yaw)
+                if self._view_angle_limit > 0.001:
```

Comment:
why 0.001? 0 seems better.

Reply:
I was being too careful in case float comparison isn't accurate.  Changed to 0.